File: frontend/code_brain_ui/src/types/task.ts
export interface PlanStep {
  id: string;
  description: string;
  status: string;
}

export interface Plan {
  steps: PlanStep[];
  status: string;
}

export interface Task {
  id: string;
  description: string;
  playbook_id?: string;
  status: 'pending' | 'planning' | 'awaiting_confirmation' | 'in_progress' | 'completed' | 'failed';
  created_at: string;
  plan?: Plan;
  results?: Record<string, any>;
}

export interface TaskCreate {
  description: string;
  playbook_id?: string;
}

File: backend/code_brain_api/app/models/task.py
from datetime import datetime
from typing import List, Optional, Dict
from enum import Enum
from pydantic import BaseModel

class TaskStatus(str, Enum):
    PENDING = "pending"
    PLANNING = "planning"
    AWAITING_CONFIRMATION = "awaiting_confirmation"
    IN_PROGRESS = "in_progress"
    COMPLETED = "completed"
    FAILED = "failed"

class PlanStep(BaseModel):
    id: str
    description: str
    status: str = "pending"

class Plan(BaseModel):
    steps: List[PlanStep]
    status: str = "pending"

class Task(BaseModel):
    id: str
    description: str
    playbook_id: Optional[str] = None
    status: TaskStatus
    created_at: datetime
    plan: Optional[Plan] = None
    results: Optional[Dict] = None

class TaskCreate(BaseModel):
    description: str
    playbook_id: Optional[str] = None
